index.js

```diff
@@ -356,6 +356,31 @@
     ($.Undefined  = type0('Undefined'))
   ];
 
+  //  ValidNumber :: Type
+  $.ValidNumber = $.NullaryType(
+    'sanctuary-def/ValidNumber',
+    function(x) { return $.Number.test(x) && !isNaN(x); }
+  );
+
+  //  FiniteNumber :: Type
+  $.FiniteNumber = $.NullaryType(
+    'sanctuary-def/FiniteNumber',
+    function(x) { return $.ValidNumber.test(x) && x !== Infinity &&
+                                                  x !== -Infinity; }
+  );
+
+  //  NonZeroValidNumber :: Type
+  $.NonZeroValidNumber = $.NullaryType(
+    'sanctuary-def/NonZeroValidNumber',
+    function(x) { return $.ValidNumber.test(x) && x !== 0 && x !== -0; }
```

Comment:
We could use `Number(x) !== 0` in place of the second and third tests. `0 === -0`, but we need `Number` if we wish to handle Number objects.